src/network/networkd-network.h

```diff
@@ -82,6 +82,19 @@ typedef struct DUID {
         uint8_t raw_data[MAX_DUID_LEN];
 } DUID;
 
+typedef struct NetworkConfigSection {
+        char *filename;
+        unsigned filename_size;
+
+        unsigned line;
```

Comment:
Hmm, why do we need filename_size? the string is NUL terminated anyway. Also, I'd recommend a simple optimization. Let's define this like this:

```c
typedef struct ConfigSection {
        unsigned line;
        char filename[];
} ConfigSection;
```

i.e. let's embedd the filename in the structure, so that we only need a single memory allocation. For allocation this would then look like this:

```
ConfigSection *cs;
…
cs = malloc(offsetof(ConfigSection, filename) + strlen(filename) + 1);
…
```